capsules/src/virtual_spi.rs

```diff
@@ -141,7 +186,43 @@ impl<'a, SPI: hil::spi::SpiMaster> hil::spi::SPIMasterDevice for SPIMasterDevice
         self.rxbuffer.replace(read_buffer);
         self.operation.set(Op::ReadWriteBytes(len));
         self.mux.do_next_op();
-
         true
     }
+
+    fn set_polarity(&self, cpol: hil::spi::ClockPolarity) {
+        self.operation.set(Op::SetPolarity(cpol));
+        self.mux.do_next_op();
+    }
+
+    fn set_phase(&self, cpal: hil::spi::ClockPhase) {
+        self.operation.set(Op::SetPhase(cpal));
+        self.mux.do_next_op();
+    }
+
+    fn set_rate(&self, rate: u32) {
+        self.operation.set(Op::SetRate(rate));
+        self.mux.do_next_op();
+    }
+
+    fn get_polarity(&self) -> hil::spi::ClockPolarity {
+        self.mux.spi.get_clock()
+    }
+
+    fn get_phase(&self) -> hil::spi::ClockPhase {
+        self.mux.spi.get_phase()
+    }
+
+    fn get_rate(&self) -> u32 {
+        self.mux.spi.get_rate()
+    }
+
+    fn send_byte(&self, val: u8) {
+        self.operation.set(Op::SendByte(val));
+        self.mux.do_next_op();
+    }
+
+#[allow(unused_variables)]
```

Comment:
Is the `unused_variables` annotation actually needed? Either way it needs to be intended.